WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs

```diff
@@ -402,20 +402,10 @@ public Task<RoundStateResponse> GetStatusAsync(RoundStateRequest request, Cancel
 
 	private void CheckCoinIsNotBanned(OutPoint input)
 	{
-		if (Prison.TryGet(input, out var inmate))
+		var banningTime = Prison.GetBanTimePeriod(input);
+		if (banningTime.Includes(DateTimeOffset.UtcNow))
 		{
-			DateTimeOffset bannedUntil;
-			if (inmate.Punishment == Punishment.LongBanned)
-			{
-				bannedUntil = inmate.Started + Config.ReleaseUtxoFromPrisonAfterLongBan;
-				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.InputLongBanned, exceptionData: new InputBannedExceptionData(bannedUntil));
-			}
-
-			if (!Config.AllowNotedInputRegistration || inmate.Punishment != Punishment.Noted)
-			{
-				bannedUntil = inmate.Started + Config.ReleaseUtxoFromPrisonAfter;
-				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.InputBanned, exceptionData: new InputBannedExceptionData(bannedUntil));
-			}
+			throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.InputBanned, exceptionData: new InputBannedExceptionData(banningTime.EndTime));
 		}
```

Comment:
1. This method feels like it belongs to the Prison.
2. How come we use this method instead of just doing this in `RegisterInputCore()`
```cs
if(Prison.IsBanned(coin.Outpoint, DateTimeOffset.UtcNow))
{
	throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.InputBanned, exceptionData: new InputBannedExceptionData(banningTime.EndTime));
}
```